crates/typst/src/foundations/str.rs

```diff
@@ -535,16 +535,16 @@ impl Str {
             }
             Some(StrPattern::Regex(re)) => {
                 let s = self.as_str();
-                let mut last = 0;
+                let mut last = None; // The last match's off-by-one end.
```

Comment:
why "off-by-one"?

Reply:
Because it doesn't point to the last match's inclusive end.. but let me delete this comment.

Reply:
But doesn't it point the last match's ... normal Rust-style end? I.e. `m.end()` for the last match?

Reply:
Yeah that's what I meant by "off-by-one", sorry for the confusion. https://docs.rs/regex/latest/regex/struct.Match.html#method.end said `m.end()` returns "The end of the match corresponds to the byte _immediately following_ the last byte in the match" (emphasis mine).